### ISSUE TITLE
HLS installed using ghcup says (GHC: 8.10.7)

### ISSUE THREAD

**Issue by jsynacek** [haskell/haskell-language-server#3562]
### Your environment
Archlinux, but also a Ubuntu 22.04 container. It doesn't seem to be specific to the environment.

Which OS do you use?
Archlinux

Which version of HLS do you use and how did you install it?
HLS 1.10.0.0

Have you configured HLS in any way (especially: a `hie.yaml` file)?
No.

### Steps to reproduce
When HLS 1.10.0.0 is installed and set using `ghcup`, running `haskell-language-server-wrapper --version` will say GHC: 8.10.7. It should say (and said in 1.9.1.0) GHC: 9.4.4 in my case.

```
$ ghcup list
...
✗  ghc   8.10.7   base-4.14.3.0
...
✔✔ ghc   9.4.4    base-4.17.0.0
...
✔✔ cabal 3.10.1.0 latest                               
...
✗  hls   1.10.0.0                                       
...

$ ghcup install hls 1.10.0.0
...
<success>

$ ghcup set hls 1.10.0.0
...
<success>

$ haskell-language-server-wrapper  --version
haskell-language-server version: 1.10.0.0 (GHC: 8.10.7) (PATH: /home/jsynacek/.ghcup/hls/1.10.0.0/lib/haskell-language-server-1.10.0.0/bin/haskell-language-server-wrapper)
```
Notice it says `8.10.7` in the output.

If I do the same thing with HLS 1.9.1.0, it works as expected:
```
$ ghcup install hls 1.9.1.0
...
$ ghcup set hls 1.9.1.0
...
$ haskell-language-server-wrapper  --version
haskell-language-server version: 1.9.1.0 (GHC: 9.4.4) (PATH: /home/jsynacek/.ghcup/hls/1.9.1.0/lib/haskell-language-server-1.9.1.0/bin/haskell-language-server-wrapper)

```

### Expected behaviour
It should say `GHC: 9.4.4` in the version output.

### Actual behaviour
It says `GHC: 8.10.7` in the output.

**Comment by fendor** [comment]
Hi, thank you for your bug report!

GHCup installs one HLS binary per GHC installation. The wrapper then decides, based on the GHC of your project, which HLS version shall be launched.
The message shows with which GHC version the wrapper has been compiled. 
In this particular case, GHC-8.10.7 has been used to compile the wrapper.
However, this should not matter to you, the wrapper should still launch the correct version, independent of which GHC version it has been compiled with.
In other words, everything should be completely fine.

To the question of why the behaviour has changed, I defer to @hasufell or @wz1000, it is likely because of some technicality.

**Comment by hasufell** [comment]
There were portability issues with e.g. 9.4.x built binaries that rendered the wrapper broken, not just the hls binary (I think it was actually glibc/bindist issues, though).

So for the wrapper we'll always choose the oldest GHC. It's a bet and mostly irrelevant to the end user.

So this bug report is really not a bug and can be closed as invalid.

**Comment by jsynacek** [comment]
Ok, thank you for the quick answers!
